src/main/java/org/mockito/internal/debugging/LocationFactory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2007 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.debugging;
+
+import org.mockito.invocation.Location;
+
+import java.io.Serializable;
+
+public final class LocationFactory {
+    private static final String PROPERTY = "mockito.locationFactory.disable";
+    private static final Factory factory;
+
+    static {
+        Factory lFactory;
```

Comment:
I assigned to lFactory because the compiler was unable to determine in the catch case that the variable wasn't already assigned to before the throw. I've moved it into a method from the `static {}` which can return and so avoid the boilerplate.